### ISSUE TITLE
Update-Package in conjunction with TFS is extremely slow

### ISSUE THREAD

**Issue by andrelombaard** [NuGet/Home#1393]
I have a Visual Studio 2015 solution containing 38 projects and I want to update all the installed packages (Third Party packages and in house packages) by executing the Update-Package command. The update is at this point running more than 5 hours. From the Package Manager Console I can see that the packages are being updated.

My manager mentioned that he previously had a similar problem on a smaller project, he removed the project from source control, updated the packages (Which was almost instant) and added the project back to source control. Is there any reason that TFS might cause slow updates? Or is this problem related to something else?

**Comment by yishaigalatzer** [comment]
Hard to tell without seeing your project, but yes TFS can definitely slow things down.

NuGet calls visual studio IDE DTE commands like AddReference, which in turn causes visual studio to add the file by the project system. and if it is in TFS it might try to go online and add it. If you have lots of projects and lots of packages and lots of files this could be a cause.

It might also be an issue with our code, we are looking at another issue where removing content (not necessarily references), we are a lot slower than we used to be.

Last - we just released nuget 3.2, I would be interested to know what version of NuGet are you using?

**Comment by andrelombaard** [comment]
Hi, thank you for your reply. I noticed that one of our developers added the packages folder to source control, no idea why this was done. I removed the folder from TFS but I can see that there are also some direct references made to some of the dll's in the packages folder which will also cause some problems (although not related to speed).

I'm was using version 3.1 but updated yesterday morning to version 3.2, will fix all the problems on our side and revert back.

**Comment by andrelombaard** [comment]
Good day, we managed to get the update down to an acceptable time, just to give you some background, we have approximately 110 c# projects that we create into Nuget packages and host on our own local Nuget server. It is impossible for us to manually create the Nuget packages for each one of these projects, so we wrote an application that will do the work for us. We realized that this tool is adding all the dll references in the bin folder to the <files> node and that all these dll files are added to the package. This is not correct, because the references are actually part of dependent packages and can be included by installing the appropriate Nuget packages. We fixed the problem which reduced the size of our Nuget packages drastically, by doing this we got the time down to 25 minutes. 

We also ran the Update-Package command with the -IgnoreDependencies and -Source flags set which further reduced the time to about 9 minutes, this is acceptable for us. I will look out for the new updates you mentioned to speed up the process.

Thank you for your friendly reply
